### PR TITLE
SIGTERM handler

### LINKED ISSUE #5: Fan control should be reset to "automatic" when script receives SIGTERM
Currently, when stopping the systemd service fan control remains in whatever mode the script was last in. If this is "manual" mode, and the fans are running at at low RPM, CPU temperatures may reach critical levels during boot / inside BIOS, where low-power states are not maintained. The script should probably make a best effort to return to "automatic" mode when it exits.

I can submit a PR when I get a chance, just logging this for the time being. Thank you!

### PR DESCRIPTION
**Pull request by NULLx76**
This PR adds a handler for SIGTERM to set all fan controls back to automatic.

I've only tested this with one host but it seems to work nicely.

Closes #5

### CODE REVIEW

File: fan_control.py [hunk 1]
```diff
@@ -226,7 +227,21 @@ def main():
...
 
+
+def graceful_shutdown(signalnum, frame):
+    strsignal = signal.strsignal(signalnum)
+    print(f"setting hosts back to automatic fan control because we received the signal: {strsignal}")
+
+    for host in config['hosts']:
+        set_fan_control("automatic", host)
+
+    sys.exit(128 + signalnum)
```
Comment (nmaggioni) [inline fan_control.py]: Exiting with code `0` may play nicer with init systems, this is a clean termination after all.

**Review by nmaggioni — CHANGES_REQUESTED** [review]
Thanks for your contribution :+1:

File: fan_control.py (hunk 1 above)
Comment (NULLx76) [reply fan_control.py]: Yeah that's true, I'll change it.

**Review by nmaggioni — APPROVED** [review]
I'll take care of backporting some details to Python 3.7, since most systems that this script will be ran on still don't have official access to Python 3.8.

**Pull request merged**